src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/CSharp/TypeAccessibilityCodeActionProviderTest.cs

```diff
@@ -0,0 +1,396 @@
+                            Code = new DiagnosticCode("CS0246")
+                        }
+                    )
+                }
+            };
+
+            var location = new SourceLocation(0, -1, -1);
+            var context = CreateRazorCodeActionContext(request, location, documentPath, contents, new SourceSpan(0, 0));
+            context.CodeDocument.SetFileKind(FileKinds.Legacy);
+
+            var provider = new TypeAccessibilityCodeActionProvider();
+            var csharpCodeActions = Array.Empty<RazorCodeAction>();
+
+            // Act
+            var results = await provider.ProvideAsync(context, csharpCodeActions, default);
+
+            // Assert
+            Assert.Empty(results);
+        }
+
+        [Fact]
+        public async Task Handle_ValidDiagnostic_InvalidCodeAction_ReturnsEmpty()
+        {
+            // Arrange
+            var documentPath = "c:/Test.razor";
+            var contents = "@code { Path; }";
+            var request = new CodeActionParams()
+            {
+                TextDocument = new TextDocumentIdentifier(new Uri(documentPath)),
+                Range = new Range(),
+                Context = new CodeActionContext()
+                {
+                    Diagnostics = new Container<Diagnostic>(
+                        new Diagnostic()
+                        {
+                            Severity = DiagnosticSeverity.Error,
+                            Code = new DiagnosticCode("CS0132")
+                        },
+                        new Diagnostic()
+                        {
+                            Severity = DiagnosticSeverity.Error,
+                            Code = new DiagnosticCode("CS0246"),
+                            Range = new Range(
+                                new Position(0, 8),
+                                new Position(0, 12)
+                            )
+                        },
+                        new Diagnostic()
+                        {
+                            Severity = DiagnosticSeverity.Error,
+                            Code = new DiagnosticCode("CS0183")
+                        }
+                    )
+                }
+            };
+
+            var location = new SourceLocation(0, -1, -1);
+            var context = CreateRazorCodeActionContext(request, location, documentPath, contents, new SourceSpan(8, 4));
+            context.CodeDocument.SetFileKind(FileKinds.Legacy);
+
+            var provider = new TypeAccessibilityCodeActionProvider();
+            var csharpCodeActions = new[] {
+                new RazorCodeAction()
+                {
+                    Title = "System.IO.OneThing"
+                },
+                new RazorCodeAction()
+                {
+                    Title = "System.IO.SomethingElse"
+                }
+            };
+
+            // Act
+            var results = await provider.ProvideAsync(context, csharpCodeActions, default);
```

Comment:
It might be helpful to the uninitiated to call out in a comment exactly what's invalid about these CodeActions.

Reply:
Given the tests below I assume it's that the namespaces are bad?

Reply:
https://github.com/dotnet/aspnetcore-tooling/pull/2525/files/74d568462f84671a667dde2803706beae41d206d#r492218640